Bio/Align/_pairwisealigner.c

```diff
@@ -5999,8 +5999,13 @@ exit: \
      * expanded nodes, lower_bound contains the global lower_bound, a and b \
      * contain the lower bounds for the current cell. ch contains the types of \
      * the potential children */ \
-    int type_total = 1, new_type, npA, npB; \
-    double new_score, new_lower, new_upper, next_lower, next_upper; \
+    int type_total = 1; \
+    /* The initial values for new_type, npA, npB, new_score, new_lower, \
+    new_upper, next_lower, and next_upper don't mean anything; they're never \
```

Comment:
Are you absolutely sure that there is no situation in which these are used before initialization?
For example, `new_upper` is used in the `while` statement at the end. It is.a very long block, and it is not immediately clear to me that there is no branch where `new_upper` may not be defined.

Reply:
Before that loop condition there's an if-statement that checks if the queue is empty. If the queue isn't empty, then `new_upper` will be set/initialized right before it is used in the loop condition. If the queue is empty, then the loop breaks and the loop condition won't be run. Similarly, `new_type` is accessed and not set in the 2nd child and 3rd child code, but that code can only run after the 1st child or queue-popping code is run which sets `new_type`.

Also, these variables are declared without initialization in the original [Chakraborty code](https://www.isical.ac.in/~bioinfo_miu/FOGSAA.htm) this is based off.